### PR TITLE
Fix writing and reading between words

### LINKED ISSUE #4: Fix mmio functions
reading a byte/2 bytes inside a word does not work

### PR DESCRIPTION
**Pull request by felixauringer**
closes #4

### CODE REVIEW

File: kernel/mmio.c [hunk 1]
```diff
@@ -19,52 +19,129 @@
...
 
 #include "mmio.h"
-#include <stdio.h>
 
 uint8_t
 read8(uint32_t address)
-{
-	return (uint8_t) (read32(address-(address % 4)) >> ((address % 4) * 8));
-}
+  {
```
Comment (sirkrypt0) [inline kernel/mmio.c]: I think those brackets are indented one to far. To me it seems like they don't have to be indented, according to Andreas' style

**Review by sirkrypt0 — CHANGES_REQUESTED** [review]
nice

File: kernel/mmio.c (hunk 1 above)
Comment (felixauringer) [reply kernel/mmio.c]: You're right, the method-brackets are not indented as in `for`, `while`, ...
I corrected that.

**Review by sirkrypt0 — APPROVED** [review]

**Review by shm0sby — APPROVED** [review]
I think we should think about some kind of testing for such functions? 🤔 
Anyway it's fine. :)

**Comment by felixauringer** [comment]
> I think we should think about some kind of testing for such functions? 
> Anyway it's fine. :)

That's a good idea, we should talk about it in the next meeting!

**Pull request merged**